react/Labs/PasswordInput/helpers.js

```diff
@@ -0,0 +1,56 @@
+// Return given password srength as an object {percentage, label}
+export const getStrength = password => {
+  if (!password && password !== '') {
+    throw new Error('password parameter is missing')
+  }
+  if (!password.length) {
+    return { percentage: 0, label: 'weak' }
+  }
+
+  const charsets = [
+    // upper
+    { regexp: /[A-Z]/g, size: 26 },
+    // lower
+    { regexp: /[a-z]/g, size: 26 },
+    // digit
+    { regexp: /[0-9]/g, size: 10 },
+    // special
+    { regexp: /[!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?]/g, size: 30 }
+  ]
```

Comment:
Those could be extracted from the function. This way we do not create new regexps at each run.